proto/astarte_msg_hub.proto

```diff
@@ -0,0 +1,118 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+syntax = "proto3";
+
+package org.astarteplatform.msghub;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+service MessageHub {
+  rpc Attach(Node) returns (stream NodeMessage) {}
+  rpc Send(NodeMessage) returns (google.protobuf.Empty){}
+  rpc Detach(Node) returns (google.protobuf.Empty){}
+}
+
+message Node {
+  string id = 1;
+  NodeIntrospection introspection = 2;
+}
+
+message NodeIntrospection {
+  repeated Interface interfaces = 1;
+}
+
+message Interface {
+  string name = 1;
+  int32 major = 2;
+  int32 minor = 3;
+}
+
+message NodeMessage{
+  Interface interface = 1;
+  string path = 2;
+  AstarteAggregation astarte_aggregation = 3;
```

Comment:
I feel like this is redundant, the aggregation can be inferred from the astarte type.